### PR TITLE
feat: add execInteractiveCmd

### PR DESCRIPTION
**Pull request by mdonnalley**
Adds `execInteractiveCmd` for executing interactive commands in NUTs

### Usage

```typescript
import * as path from 'path';
import { TestSession, execInteractiveCmd, Interaction } from '@salesforce/cli-plugins-testkit';
import { env } from '@salesforce/kit';

describe('dev generate plugin NUTs', () => {
  let session: TestSession;

  before(async () => {
    env.setString('TESTKIT_EXECUTABLE_PATH', path.join(process.cwd(), 'bin', 'dev'));
    session = await TestSession.create({});
  });

  after(async () => {
    await session?.clean();
  });

  it('should do interactive things', async () => {
    const result = await execInteractiveCmd(
      'dev generate plugin',
      {
        // these keys are used to match the questions asked
        'internal Salesforce team': Interaction.Yes,
        'name of your new plugin': ['plugin-awesome', Interaction.ENTER],
        'description for your plugin': ['a description', Interaction.ENTER],
        'Select the existing "sf" commands you plan to extend': [
          Interaction.SELECT,
          Interaction.DOWN,
          Interaction.SELECT,
          Interaction.ENTER,
        ],
      },
      { cwd: session.dir, ensureExitCode: 0 }
    );
  });
});

```

@W-11787546@

### CODE REVIEW

File: src/execCmd.ts [hunk 1]
```diff
@@ -303,3 +303,125 @@ export function execCmd<T = Collection>(
     }
   }
 }
+
+function toString(arrOrString: string | string[]): string {
```
Comment (peternhale) [inline src/execCmd.ts:307]:
``` suggestion
function toString(arrOrString: Many<string>): string {
```

Comment (mshanemc) [reply src/execCmd.ts]: ooh, `Many` is an unfortunate name for that

File: src/execCmd.ts [hunk 2]
```diff
@@ -303,3 +303,159 @@ export function execCmd<T = Collection>(
...
 }
+
+function toString(arrOrString: Many<string>): string {
+  if (Array.isArray(arrOrString)) {
+    return arrOrString.join('');
+  }
+  return arrOrString;
+}
+
+function toArray(arrOrString: Many<string>): string[] {
```
Comment (mshanemc) [inline src/execCmd.ts:314]: use `ensureArray` from kit

**Review by mshanemc — APPROVED** [review]

**Comment by mshanemc** [comment]
QA Notes:

❓  I think this should accept a type `await execInteractiveCmd<MyResultType>()` for consistency with the other execs and to make it more TS friendly.  Unless we decided that interactive commands can never have json (even some of the confirmation prompt stuff might benefit)

👎🏻 execCmd (non-interactive) doesn't require you to pass in cwd--it runs by default in the project dir.  I don't want this one to behave so differently (and your example uses session.dir instead of `session.project.dir`)
- I get that some commands like the dev/generate need to run from outside a project, but I'm guessing *most* interactive commands will be project related?

---
unrelated to this PR, but probably related to oclif parser changes for multiple?

```ts
    icon: Flags.integer({
      char: 'i',
      required: true,
      min: 1,
      max: 100,
      default: 1,
      summary: messages.getMessage('flags.icon.summary'),
      description: messages.getMessage('flags.icon.description'),
    }),
  };

  public async run(): Promise<GenerateTabResult> {
    const { flags } = await this.parse(GenerateTab);
    const tabPath = path.join(flags.directory, `${flags.object}.tab-meta.xml`);

    const tab: GenerateTabResult['tab'] = {
      customObject: true,
      motif: tabDefs.find((tabDef) => tabDef.includes(`Custom${flags.icon}:`)),

```
`Invalid type "number | number[]" of template literal expression.`

File: src/execCmd.ts (hunk 2 above)
Comment (mdonnalley) [reply src/execCmd.ts:314]: `ensureArray` throws an error if the provided value isn't an array, which isn't what I'm doing here

**Comment by mdonnalley** [comment]
@mshanemc I don't think it's possible to put a type on the return. The prompts go to stdout which makes parsing JSON impossible

**Review by mshanemc — APPROVED** [review]

**Pull request merged**